File: worker/structs/report.go
package structs

import (
	"gopkg.in/mgo.v2/bson"
	"time"
)

type TotalComment struct {
	Main string `bson:"main"`
	Sub  string `bson:"sub"`
}

type Endurance struct {
	Main    string `bson:"main"`
	Sub     string `bson:"sub"`
	Percent int    `bson:"percent"`
}

type Report struct {
	Id           bson.ObjectId `bson:"_id,omitempty"`
	Monthly      bool          `bson:"isMonthly"`
	Name         string        `bson:"name"`
	GenerateAt   time.Time     `bson:"generateAt"`
	Result       []int         `bson:"result"`
	Total        int           `bson:"total"`
	TotalComment TotalComment  `bson:"totalCommnet"`
	Average      int           `bson:"average"`
	Count        int           `bson:"count"`
	Fail         [4]int        `bson:"fail"`
	Endurance    Endurance     `bson:"endurance"`
}

type ReportTemplete struct {
	Id     bson.ObjectId `bson:"_id,omitempty"`
	UserId bson.ObjectId `bson:"userId"`
	GoalId bson.ObjectId `bson:"goalId"`
	report []Report      `bson:"report"`
}

File: server/models/report.js
/**
 * Created by juwoong on 2015. 9. 3..
 */
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var Report = new Schema({
    isMonthly       : Boolean,
    name            : String,
    generateAt      : Date,
    result          : [Number],
    total           : Number,
    totalComment    : {
        main        : String,
        sub         : String
    },
    average         : Number,
    count           : Number,
    fail            : [Number],
    endurance       : {
        main        : String,
        sub         : String,
        percent     : Number
    }
});

var ReportTemplete = new Schema({
    userId          : {type : Schema.Types.ObjectId, ref : 'user'},
    goalId          : {type : Schema.Types.ObjectId, ref : 'goal'},
    report          : [Report]
});

module.exports.reportTemplete = mongoose.model("reportTemplete", ReportTemplete);
module.exports.report = mongoose.model("report", Report);